Tests/Functional/Indexer/NodeIndexerDimensionsTest.php

```diff
@@ -0,0 +1,270 @@
+            ->sortDesc('title')
+            ->execute();
+
+        /** @var QueryResultInterface $resultDefault */
+        static::assertCount(1, $resultDefault, 'The result should have 2 items');
+        static::assertEquals(1, $resultDefault->count(), 'Count should be 2');
+
+        $resultDe = $queryBuilder
+            ->query($this->siteNodeDe)
+            ->log($this->getLogMessagePrefix(__METHOD__))
+            ->nodeType('Neos.NodeTypes:Page')
+            ->sortDesc('title')
+            ->execute();
+
+        /** @var QueryResultInterface $resultDe */
+
+        static::assertCount(4, $resultDe, 'The result should have 4 items');
+        static::assertEquals(4, $resultDe->count(), 'Count should be 4');
+
+        $resultFr = $queryBuilder
+            ->query($this->siteNodeFr)
+            ->log($this->getLogMessagePrefix(__METHOD__))
+            ->nodeType('Neos.NodeTypes:Page')
+            ->sortDesc('title')
+            ->execute();
+
+        /** @var QueryResultInterface $resultFr */
+
+        static::assertCount(3, $resultFr, 'The result should have 3 items');
+        static::assertEquals(3, $resultFr->count(), 'Count should be 3');
+    }
+
+
+    /**
+     * add test data that moreless look like this:
+     * -root-[en_US|de|fr]
+     *   |- site-[en_US|de|fr]
+     *     |- document2-[de|fr]
+     *       |- document3-de
+     *       |- document4-[de|fr]
+     *
+     * @throws NodeExistsException
+     * @throws NodeTypeNotFoundException
+     * @throws StopActionException
+     * @throws \Flowpack\ElasticSearch\ContentRepositoryAdaptor\Exception
+     * @throws ApiException
+     * @throws StopCommandException
+     */
+    protected function createNodesForLanguageDimensions(): void
+    {
+        $defaultLanguageDimensionContext = $this->contextFactory->create([
+            'workspaceName' => 'live',
+            'dimensions' => ['language' => ['en_US']],
+            'targetDimensions' => ['language' => 'en_US']
+        ]);
+        $deLanguageDimensionContext = $this->contextFactory->create([
+            'workspaceName' => 'live',
+            'dimensions' => ['language' => ['de']],
+            'targetDimensions' => ['language' => 'de']
+        ]);
+        $frLanguageDimensionContext = $this->contextFactory->create([
+            'workspaceName' => 'live',
+            'dimensions' => ['language' => ['fr']],
+            'targetDimensions' => ['language' => 'fr']
+        ]);
+
+        $rootNode = $defaultLanguageDimensionContext->getRootNode();
+        $this->siteNodeDefault = $rootNode->createNode('root', $this->nodeTypeManager->getNodeType('Neos.NodeTypes:Page'));
+        $this->siteNodeDefault->setProperty('title', 'root-default');
+        $this->siteNodeDe = $deLanguageDimensionContext->adoptNode($this->siteNodeDefault, true);
+        $this->siteNodeDe->setProperty('title', 'root-de');
+        $this->siteNodeFr = $deLanguageDimensionContext->adoptNode($this->siteNodeDefault, true);
```

Comment:
Looks like $frLanguageDimensionContext should be used here?

Reply:
You are right, that is a mistake